src/Evolution/Systems/GeneralizedHarmonic/Initialize.hpp

```diff
@@ -146,35 +146,38 @@ struct InitializeDampedHarmonicRollonGauge {
     // compute initial-gauge related quantities
     const auto& mesh = db::get<domain::Tags::Mesh<Dim>>(box);
     const size_t num_grid_points = mesh.number_of_grid_points();
-    const auto& lapse = get<gr::Tags::Lapse<DataVector>>(box);
-    const auto& dt_lapse = get<::Tags::dt<gr::Tags::Lapse<DataVector>>>(box);
-    const auto& deriv_lapse = get<
-        ::Tags::deriv<gr::Tags::Lapse<DataVector>, tmpl::size_t<Dim>, frame>>(
-        box);
-    const auto& shift = get<gr::Tags::Shift<Dim, frame, DataVector>>(box);
-    const auto& dt_shift =
-        get<::Tags::dt<gr::Tags::Shift<Dim, frame, DataVector>>>(box);
-    const auto& deriv_shift =
-        get<::Tags::deriv<gr::Tags::Shift<Dim, frame, DataVector>,
-                          tmpl::size_t<Dim>, frame>>(box);
-    const auto& spatial_metric =
-        get<gr::Tags::SpatialMetric<Dim, frame, DataVector>>(box);
-    const auto& trace_extrinsic_curvature =
-        get<gr::Tags::TraceExtrinsicCurvature<DataVector>>(box);
-    const auto& trace_christoffel_last_indices =
-        get<gr::Tags::TraceSpatialChristoffelFirstKind<Dim, frame, DataVector>>(
+
+    const auto& spacetime_metric =
+        db::get<gr::Tags::SpacetimeMetric<Dim, Frame::Inertial, DataVector>>(
             box);
+    const auto& pi =
+        db::get<GeneralizedHarmonic::Tags::Pi<Dim, Frame::Inertial>>(box);
+    const auto& phi =
+        db::get<GeneralizedHarmonic::Tags::Phi<Dim, Frame::Inertial>>(box);
+
+    const auto spatial_metric = gr::spatial_metric(spacetime_metric);
+    const auto inverse_spatial_metric =
+        determinant_and_inverse(spatial_metric).second;
+    const auto shift = gr::shift(spacetime_metric, inverse_spatial_metric);
+    const auto lapse = gr::lapse(shift, spacetime_metric);
+    const auto inverse_spacetime_metric =
+        gr::inverse_spacetime_metric(lapse, shift, inverse_spatial_metric);
+    tnsr::abb<DataVector, Dim, Frame::Inertial> da_spacetime_metric{};
+    GeneralizedHarmonic::spacetime_derivative_of_spacetime_metric(
+        make_not_null(&da_spacetime_metric), lapse, shift, pi, phi);
+    // H_a=-Gamma_a
```

Comment:
Just a suggestion: it might be safer / more useful if instead of computing H_a here, you move this code as an overload of `gauge_source()`, and then call it here with `spacetime_metric, pi, phi` as inputs.

Reply:
So I had thought about this while writing the code and I worry that doing so will lead to confusion with the gauge source as computed from the damped harmonic H functions, since the damped harmonic H functions are what you should pretty much always use unless you want to check the constraint or initialize H_a. What are your thoughts?

Reply:
I slightly prefer the version as Nils coded it

Reply:
> So I had thought about this while writing the code and I worry that doing so will lead to confusion with the gauge source as computed from the damped harmonic H functions, since the damped harmonic H functions are what you should pretty much always use unless you want to check the constraint or initialize H_a. What are your thoughts?

Agreed that this would lead confusion if trying to compute the intended H_a, which should preferably be computed through damped harmonic functions. I meant more like an overload of [this](https://github.com/sxs-collaboration/spectre/blob/fa855c03790658c3b67039da4ad8fe3bba078678/src/PointwiseFunctions/GeneralRelativity/ComputeGhQuantities.hpp#L919) which computes the gauge source *implicit* from evolved variables. Having said that, I confess I can't think of many uses cases for this overload so keep the current code. Its clean as it is.